src/core/server/src/process.rs

```diff
@@ -83,12 +86,13 @@ where
         let workers = workers.spawn();
         let cloned_signal_tx = signal_tx.clone();
 
-        //Signal handlers
-        ctrlc::set_handler(move || {
-            //Handle SIGINT by making a best attempt at signalling shutdown via admin thread.
-            Process::shutdown_signal(&cloned_signal_tx);
-        })
-        .expect("Error setting Ctrl-C handler");
+        // NOTE: Signal handler join handle is not taken ownership of by [Process] as it's
+        // considered something that has the same lifetime as the actual OS process as a whole
+        // and there aren't any current use cases for blocking on join()'ing the thread
+        // if we want to dynamically rebind signal handlers in the future we should reconsider this
+        let _signal_handler = std::thread::Builder::new()
+            .name(format!("{THREAD_PREFIX}_signal_handler"))
```

Comment:
That's a very long thread name. It's probably going to be truncated in most places we'll see it. I'd suggest just `{THREAD_PREFIX}_signal` instead.